sdk/eventhub/event-hubs/src/eventProcessor.ts

```diff
@@ -114,6 +114,19 @@ export class EventProcessor {
       partitionContext,
       new CheckpointManager()
     );
+    if (partitionProcessor.initialize && typeof partitionProcessor.initialize !== "function") {
```

Comment:
We should only be validating `processEvents` and `processError` here. We currently say that `initialize` and `close` are optional, so we should not throw an error if they are missing.

Edit: The code in this `start` method is going to change a lot once we add multiple partition processing, so actually I'm not too worried about how this method looks. I just want to make sure we will work if `initialize` and `close` are not provided.

Reply:
We are not throwing if `partitionProcessor.initialize` and `partitionProcessor.close` are missing.

Here I'm checking if user provides these optional methods then only I'm checking the type.

```javascript
if (partitionProcessor.initialize && typeof partitionProcessor.initialize !== "function") {
      throw new TypeError("'initialize' must be of type 'function'.");
    }
 if (partitionProcessor.close && typeof partitionProcessor.close !== "function") {
      throw new TypeError("'close' must be of type 'function'.");
    }
```

Reply:
Yup, you're right! I misread what this was doing 😢